android/hybridcommon/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt

```diff
@@ -296,27 +306,23 @@ fun purchaseProduct(
                 }
 
                 // Add ons
-                if (!addOnStoreProducts.isNullOrEmpty()) {
-                    val addOns = addOnStoreProducts.mapNotNull { addOnMap ->
-                        val addOnProductIdentifier = addOnMap["productIdentifier"] as? String ?: return@mapNotNull null
-                        val rawType = addOnMap["type"] as? String ?: return@mapNotNull null
-                        val addOnType = mapStringToProductType(rawType)
-                        var storeProduct = storeProductForProductId(
-                            productId = addOnProductIdentifier,
-                            type = addOnType,
-                            basePlanId = null,
-                            storeProducts = storeProducts,
-                        )
-                        if (storeProduct == null) { return@mapNotNull null }
-                        storeProduct = copyPresentedOfferingContextFromMapIfAvailable(
```

Comment:
Removed the logic to preserve the presenting offering context in the add-ons since the backend only considers the context of the base item in a subscription with add-ons